torch/csrc/autograd/python_engine.cpp

```diff
@@ -208,6 +211,7 @@ PyObject *THPEngine_run_backward(THPEngine *self, PyObject *args, PyObject *kwar
       }
       THPUtils_assert(input_var->cdata.requires_grad(),
           "One of the differentiated Variables does not require grad");
+      if (allow_unreachable && !grad_fn) continue;
```

Comment:
it could be unreachable and have a grad_fn right?
For example:

```
a = b * c
d = c * e
torch.autograd.grad(a, d, ...)
```

Reply:
Yes. Both cases are tested